src/Illuminate/Database/Eloquent/Relations/BelongsToMany.php

```diff
@@ -324,6 +360,15 @@ public function getPivotClass()
     public function using($class)
     {
         $this->using = $class;
+        $this->pivotTable = (new $class)->getTable();
+        $this->pivotQuery = $this->newPivotStatement();
+
+        //if the method is called after the constructor ran, which possibly sets the query without the needed parameters
+        //for "performJoins", reinitialize the query and call addConstraints
+        if ($this->query) {
+            $this->query = $this->related->newModelQuery();
+            $this->addConstraints();
+        }
```

Comment:
This feels pretty hacky and will lose any custom `where` that was added in the relationship method.

Reply:
Removing the "late" using-call would have been much cleaner but would have changed the interface.

Reply:
Do you have any comments regarding breaking `wheres` that are added during the relationship definition?

Reply:
Or really, any other type of query element... orderBys, etc.

Reply:
Unless i miss something, all the other wheres, orderBys and such should not be removed if the the pivot is configured during the constructor. 

It should even work if it is used like `return $this->belongsToMany()->using()->where()->orderBy()`

Reply:
I removed this part, since it is no longer necessary. 
I already moved the creation of the join / joinSub out of the constructor call-chain and delayed that part to the point where the query will actually be executed. 
All tests are still passing, so i'm confident i didn't break the new features again ;)